packages/contracts/src/strategies/BaseStrategy.sol

```diff
@@ -266,11 +267,13 @@ abstract contract BaseStrategy is
             uint256 debtPayment
         )
     {
-        uint256 amountFreed = _liquidateAllPositions();
-        if (amountFreed < outstandingDebt) {
-            loss = outstandingDebt - amountFreed;
-        } else if (amountFreed > outstandingDebt) {
-            profit = amountFreed - outstandingDebt;
+        _liquidateAllPositions();
+        uint256 freeBalance = _underlyingBalance();
+
+        if (freeBalance < outstandingDebt) {
+            loss = outstandingDebt - freeBalance;
+        } else if (freeBalance > outstandingDebt) {
+            profit = freeBalance - outstandingDebt;
```

Comment:
Please add test for this logic change

Reply:
I'm out of ideas on how to test it properly in the current logic. However this part of code has already covered with other tests including proper branching according to coverage report

Reply:
Okay, let's keep existing tests